Base/Python/slicer/packaging/installer.py

```diff
@@ -0,0 +1,400 @@
+"""Functions and classes related to installation and uninstallation of Python packages."""
+
+import contextlib
+import importlib.resources
+import importlib.util
+import inspect
+import io
+import logging
+import shlex
+import subprocess
+import sys
+import textwrap
+import types
+from contextlib import AbstractContextManager, ExitStack
+from pathlib import Path
+from typing import Optional, Union
+
+import slicer.util
+
+__all__ = [
+    "FileIdentifier",
+    "InstallationAbortedError",
+    "pip_install",
+    "pip_uninstall",
+    "register_constraints",
+]
+
+UV_ENV = {
+    "UV_PYTHON": sys.executable,
+    "UV_SYSTEM_PYTHON": "true",  # This allows uv to modify `UV_PYTHON=sys.executable`.
+    "UV_PYTHON_DOWNLOADS": "never",
+    "UV_NO_PROGRESS": "true",
+    "NO_COLOR": "true",
+    # todo Should 'UV_CACHE_DIR' be somewhere in Slicer data? Probably not.
+    # todo Should 'UV_COMPILE_BYTECODE' be enabled? Probably not.
+}
+"""
+Common environment variables passed to UV, to ensure only the correct Python environment is used.
+"""
+
+
+ResourceName: type = str
+"""
+Alias for string resource identifiers of the form ``'package:path'``, for example
+``'slicer.packaging:core-constraints.txt'``.
+
+Note this _should_ be annotated ``TypeAlias``, but that is not available in Python 3.9.
+"""
+
+
+class FileIdentifier:
+    name: str
+    identifier: ResourceName
+    caller: types.ModuleType
+
+    def __init__(
+        self,
+        name: str,
+        identifier: ResourceName,
+        caller: types.ModuleType = None,
+    ):
+        """
+        A named identifier to a Python package resource file, used to access ``requirements.txt``
+        and ``constraints.txt`` in a portable manner, with a name and context for better reporting.
+
+        :param name: A name for this resource, to be presented to the user.
+        :param identifier: The identifier to this resource, in the format ``'package:file'``, for
+          example ``'slicer.packaging:core-constraints.txt'``.
+        :param caller: The module which depends upon this resource. If not given, this is assumed
+          to be the module that called this function.
+        """
+
+        if caller is None:
+            calling_frame = inspect.currentframe().f_back
+            caller = inspect.getmodule(calling_frame)
+
+        self.name = name
+        self.identifier = identifier
+        self.caller = caller
+
+    def as_file(self) -> AbstractContextManager[Path]:
+        package_name, _, path = self.identifier.rpartition(":")
+
+        # An omitted package_name is a relative import for convenience.
+        if not package_name:
+            package_name = "."
```

Comment:
Remove this. Do not do relative import.

If the package is not provided, do a path lookup on `sys.path`. Perhaps there is something to help with this in `importlib.util`? But it is not handled by `importlib.resources`.